cli/src/main/java/com/devonfw/tools/ide/environment/EnvironmentVariablesPropertiesFile.java

```diff
@@ -184,9 +219,31 @@ public void save() {
       }
       this.modifiedVariables.clear();
     } catch (IOException e) {
-      throw new IllegalStateException("Failed to save properties to " + newPropertiesFilePath, e);
+      throw new IllegalStateException("Failed to save properties to " + this.propertiesFilePath, e);
     }
-    this.propertiesFilePath = newPropertiesFilePath;
+    this.legacyConfiguration = Boolean.FALSE;
+  }
+
+  private List<VariableLine> loadVariableLines(Path file) {
```

Comment:
Why do we need this and can't use the Properties library?

Reply:
Are you talking about `java.util.Properties`?
This does not support `export` nor does it support comments so if you do `Properties.load(...)` and then `Properties.save(...)` all the comments are lost and the order of your file is random.
I started with `java.util.Properties` since I also do not like reinventing the wheel but it is unfortunately not helpful for our problem.